programs/vote/src/vote_state/mod.rs

```diff
@@ -253,13 +253,16 @@ impl VoteState {
     ///
     ///  if commission calculation is 100% one way or other,
     ///   indicate with false for was_split
-    pub fn commission_split(&self, on: f64) -> (f64, f64, bool) {
+    pub fn commission_split(&self, on: u64) -> (u64, u64, bool) {
         match self.commission.min(100) {
-            0 => (0.0, on, false),
-            100 => (on, 0.0, false),
+            0 => (0, on, false),
+            100 => (on, 0, false),
             split => {
-                let mine = on * f64::from(split) / f64::from(100);
-                (mine, on - mine, true)
+                let on = u128::from(on);
+                let mine = on * u128::from(split) / 100u128;
+                let theirs = on * u128::from(100 - split) / 100u128;
```

Comment:
@rwalker-com Btw, is there any specific reason for changing the formula here slightly? This may cause 1 lamports to gone: https://play.rust-lang.org/?version=nightly&mode=release&edition=2018&gist=f8f4403a1b9638be9e2455e246ac40d6

If so, could you comment a bit on this pr? I'll copy that into the source code as well. :)

However, this won't cause capitalization/total-supply mismatches.

Reply:
Yes: a lamport may be lost with this version.  I was upset with the f64->u64 conversion wherein faithful conversion is capped at ~2^53.

This version is intended to catch the case where the small side gets paid 0 as a result of the integer math.

If the small side would receive zero when the split isn't 0 or 100, we don't pay either side.

Reply:
@rwalker-com Thanks for explaining! I've tried to comment the intention combining your explanation with [these new test assertions](https://github.com/solana-labs/solana/pull/10914/files#diff-3666262446ea129567c6d96960c7e23cR1574-R1578): 29c444343e

Does this make sense?